testing.py

```diff
@@ -184,11 +192,20 @@ def _work(self, title=None, authors=None, genre=None, language=None,
             genre, ignore = Genre.lookup(self._db, genre, autocreate=True)
         work.genres = [genre]
         work.random = 0.5
+
         work.editions = [primary_edition]
         primary_edition.is_primary_for_work = True
-        work.primary_edition = primary_edition
+
+        #work.primary_edition = primary_edition
+        work.set_primary_edition()
+
         if pool != None:
-            work.license_pools.append(pool)
+            # make sure the pool's presentation_edition is set, 
+            # bc loan tests assume that.
+            pool.set_presentation_edition(None)
+
+            if not work.license_pools[0]:
```

Comment:
This looks like work.license_pools is a list where the first item is `None`. I think maybe you meant to check if the list is empty? Like, `if not work.license_pools`.

The fact that this never happens (if the list was empty you'd get a crash here) indicates that it might be okay to remove this bit.